dhash/module.py

```diff
@@ -18,20 +21,36 @@
 LIMIT_SOFT = 14
 
 MAX_ATTACHMENT_SIZE = 8000
+ALLOWED_FORMATS = ("jpg", "jpeg", "png", "webp", "gif")
+
+HEADERS = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36"
+}
+
+URL_REGEX = r"(https?://[^\s]+)"
 
 
 class Dhash(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
         self.embed_cache = {}
 
+        self.allowed_urls = HashConfig.get("allowed_urls", None)
+
+        try:
+            self.allowed_urls = re.compile(self.allowed_urls)
+        except (re.error, TypeError):
+            self.allowed_urls = None
```

Comment:
Since there are only a few messages posted every minute, we can ignore this caching and load the config from the database when it is needed.

Reply:
We can, but it's not that big of a problem to cache it, since it's global value for all guilds and also I count on compiling that regex. There's no reason to trade few bytes of memory for accessing DB and CPU time.

Reply:
ok